styles/atom-ide-outline.less

```diff
@@ -83,20 +84,65 @@
   .icon {
     contain: @contain_except_size;
     display: inline-block;
-    width: 40px;
+    width: 5ex;
+    font-size: 75%;
     text-align: center;
-    vertical-align: top; // align icon with text vertically
+    vertical-align: middle; // align icon with text vertically
     font-weight: normal;
   }
 
   // syntax-variables for languge entites: https://github.com/atom/atom/blob/master/static/variables/syntax-variables.less#L32
   // atom-languageclient mapping: https://github.com/atom/atom-languageclient/blob/485bb9d706b422456640c9070eee456ef2cf09c0/lib/adapters/outline-view-adapter.ts#L270
 
+  .iconByType(@type) {
+    content: replace(@type, "^(....).*$", '"$1"'); // use the first 4 letter of the type
+  }
+  .iconByType(array) {
+    content: "arr";
+  }
+  .iconByType(constructor) {
+    content: "ctor";
+  }
+  .iconByType(field) {
+    content: "fild";
+  }
+  .iconByType(function) {
+    font-family: "symbol-icons";
+    content: "\e608";
+  }
+  .iconByType(interface) {
+    content: "intf";
+  }
+  .iconByType(module) {
+    content: "mod";
+  }
+  .iconByType(namespace) {
+    font-family: "symbol-icons";
+    content: "\e609";
+  }
+  .iconByType(number) {
+    content: "num";
+  }
+  .iconByType(package) {
+    content: "pkg";
+  }
+  .iconByType(string) {
+    content: "str";
+  }
+  .iconByType(variable) {
+    font-family: "symbol-icons";
+    content: "\e607";
+  }
+
   .styleByType(@type, @color) {
     .type-@{type} {
-      contain: @contain_except_size;
       color: @color;
-      font-size: smaller;
+      &::before {
+        .iconByType(@type);
```

Comment:
```suggestion
        .iconByType(@type);
        margin-right: 30px;  // prevents clipping of the icons once the user zooms
```

Reply:
Uh. There shouldn't be any clipping even without this. As long as "icons" are no more than 4 characters wide.

Reply:
It clips. See the above suggestion